web/src/app/(record)/defecation/_components/ui/select-defecation/color.tsx

```diff
@@ -1,34 +1,49 @@
 'use client';
+'use client';
 
+import { Controller, useFormContext } from 'react-hook-form';
 import { cn } from '@/utils/utils-cn';
-import { DEFECATION_TRY_COLOR } from '../../constants/description';
-import { useDefecation } from '../../providers/defecation-providers';
+import { DEFECATION_COLOR } from '../../constants';
+import type { DefecationFormValues } from '../../schemas';
 
-export default function Color() {
-  const { defecationState, setDefecationState } = useDefecation();
+export default function Color({
+  onColorSelect,
+}: {
+  onColorSelect?: () => void;
+}) {
+  const { control, setValue } = useFormContext<DefecationFormValues>();
 
   return (
     <div className="flex items-center justify-between gap-[12.6px] max-[398px]:gap-1">
-      {Object.entries(DEFECATION_TRY_COLOR).map(([key, value]) => (
-        <button
-          type="button"
-          onClick={() => {
-            setDefecationState({ ...defecationState, selectedColor: key });
-          }}
-          key={key}
-          className="flex flex-col items-center justify-center gap-2.5"
-        >
-          <div
-            className={cn(
-              'w-10 h-10 rounded-[10px]',
-              defecationState.selectedColor === key &&
-                'border-[1px] border-white'
-            )}
-            style={{ backgroundColor: value[1] }}
-          />
-          <div className="text-sm font-normal">{value[0]}</div>
-        </button>
-      ))}
+      <Controller
+        name="selectedColor"
+        control={control}
+        render={({ field }) => (
+          <>
+            {Object.entries(DEFECATION_COLOR).map(([key, value]) => (
```

Comment:
```suggestion
            {Object.entries(DEFECATION_COLOR).map(([key,[lebel, color]]) => (
```

사용 시 value[1,0] 이 어색하긴 하네용 object 여서 key 로 사용하면 더 명확할 것 같기는 합니다. 아니면 위 코드와 같이 변경되면 좋을 것 같아요